src/components/subscription-list/index.tsx

```diff
@@ -12,14 +12,48 @@ const SubscriptionList: React.FC = () => {
   const [message, setMessage] = useState('')
   const [messageType, setMessageType] = useState<'success' | 'error' | ''>('')
 
-  const handleSubscribe = () => {
+  const clearMessageAfterTimeout = () => {
+    setTimeout(() => {
+      setMessage('')
+      setMessageType('')
+    }, 3000)
+  }
+
+  const checkEmail = async (email: string): Promise<boolean> => {
+    const apiKey = 'e8ed0eb649msh5054f327a1dc9d7p168b47jsn142a5fd7951b'
+    const url = `https://mailcheck.p.rapidapi.com/?email=${encodeURIComponent(
+      email
+    )}`
+
+    try {
+      const response = await fetch(url, {
+        method: 'GET',
+        headers: {
+          'x-rapidapi-host': 'mailcheck.p.rapidapi.com',
+          'x-rapidapi-key': apiKey,
+        },
+      })
+
+      const data = await response.json()
+      return data.valid
```

Comment:
Perhaps we need to block not only invalid emails, but also emails flagged as disposable or blacklisted. I've tested this endpoint on Postman using an invalid email. Even though the email was incorrect, the `valid` field was `true`:

{
    "valid": true,
    "block": true,
    "disposable": true,
    "email_forwarder": false,
    "domain": "test.com",
    "text": "Should be blocked",
    "reason": "Blacklisted",
    "risk": 95,
   ...
}

I've proposed some changes on [this branch](https://github.com/vtexdocs/devportal/pull/991/files#diff-2e7cabbe31b5b2f0b15a40e29c3c9f9664794a3e5e357f8a3b997df36fda6088R38-R39) so we can test this.

Also, we will need to find a way to use environment variables, for example, so we don't expose the credentials.